### PR TITLE
ESEB-65: Add contribution exchange rate custom fields

### PR DESCRIPTION
**Pull request by olayiwola-compucorp**
## Overview
This PR adds a contribution custom group `Currency_Exchange_rates` with fields 
- Sales_Tax_Currency
- Rate_1_Unit_Tax_Currency
- Rate_1_Unit_Contribution_Currency

The value of these fields is computed when a contribution is created/updated using the Exchange Rate values in `civicrm/exchange-rate` if the contribution currency is different from the sales tax currency in the exchange rate settings.

## Before
_N/A_

## After
wqqqq

## Technical Details

The latest exchange rate is used by filtering by the exchange rate date being earlier than the contribution's receive date, and matching the conversion currency to the contribution's currency and base currency to the defined sales tax currency.

### CODE REVIEW

File: CRM/Financeextras/Upgrader.php [hunk 1]
```diff
@@ -116,16 +120,4 @@ public function upgrade_1000() {
     return TRUE;
   }
 
-  /**
-   *
-   * This upgrade creates financeextras_exchange_rate table
-   *
-   */
-  public function upgrade_1001() {
-    $this->ctx->log->info('Applying update 1001');
```
Comment (omarabuhussein) [inline CRM/Financeextras/Upgrader.php:125]: why did you remove the upgrader ?

Comment (olayiwola-compucorp) [reply CRM/Financeextras/Upgrader.php:125]: not needed,for now.

There's  `upgrade_1000()`

**Review by omarabuhussein — APPROVED** [review]
